native/src/main/java/io/ballerina/stdlib/data/jsondata/json/JsonParser.java

```diff
@@ -0,0 +1,1301 @@
+                        sm.append(ch);
+                        state = this;
+                        continue;
+                    }
+                    break;
+                }
+                sm.index = i + 1;
+                return state;
+            }
+
+        }
+
+        /**
+         * Represents the state during a string value is defined.
+         */
+        private static class StringValueState implements State {
+
+            @Override
+            public State transition(StateMachine sm, char[] buff, int i, int count) throws JsonParserException {
+                State state = null;
+                char ch;
+                for (; i < count; i++) {
+                    ch = buff[i];
+                    sm.processLocation(ch);
+                    if (ch == sm.currentQuoteChar) {
+                        sm.currentJsonNode = JsonCreator.convertAndUpdateCurrentJsonNode(sm,
+                                StringUtils.fromString(sm.value()), sm.expectedTypes.peek());
+                        state = DOC_END_STATE;
+                    } else if (ch == REV_SOL) {
+                        state = STRING_VAL_ESC_CHAR_PROCESSING_STATE;
+                    } else if (ch == EOF) {
+                        throw new JsonParserException("unexpected end of JSON document");
+                    } else {
+                        sm.append(ch);
+                        state = this;
+                        continue;
+                    }
+                    break;
+                }
+                sm.index = i + 1;
+                return state;
+            }
+        }
+
+        private void processValue() {
+            Type expType = expectedTypes.pop();
+            BString value = StringUtils.fromString(value());
+            if (expType == null) {
+                return;
+            }
+
+            currentJsonNode = JsonCreator.convertAndUpdateCurrentJsonNode(this, value, expType);
+        }
+
+        /**
+         * Represents the state during a non-string value is defined.
+         */
+        private static class NonStringValueState implements State {
+
+            @Override
+            public State transition(StateMachine sm, char[] buff, int i, int count) throws JsonParserException {
+                State state = null;
+                char ch;
+                for (; i < count; i++) {
+                    ch = buff[i];
+                    sm.processLocation(ch);
+                    if (StateMachine.isWhitespace(ch) || ch == EOF) {
+                        sm.currentJsonNode = null;
+                        sm.processValue();
+                        state = DOC_END_STATE;
+                    } else {
+                        sm.append(ch);
+                        state = this;
+                        continue;
+                    }
+                    break;
+                }
+                sm.index = i + 1;
+                sm.currentJsonNode = JsonCreator.convertAndUpdateCurrentJsonNode(sm,
```

Comment:
This may occurs issues since we refill the buffer when the character count exceed 1024

Reply:
Fixed this here https://github.com/ballerina-platform/module-ballerina-data.jsondata/pull/2/commits/72881e67ac13ff63d28c685aa88bc30440e5b093